test/e2e/tests/confirmations/transactions/speed-up-and-cancel-confirmations.spec.ts

```diff
@@ -35,8 +35,8 @@ describe('Speed Up and Cancel Transaction Tests', function () {
           // Create initial stuck transaction
           await createDappTransaction(driver, {
             value: ethInHexWei(0.1),
-            maxFeePerGas: decimalToPrefixedHex(0),
-            maxPriorityFeePerGas: decimalToPrefixedHex(0),
+            maxFeePerGas: '0x22ae4b8bcb',
+            maxPriorityFeePerGas: '0x59682f04',
```

Comment:
passing 0 here makes the tx directly to fail in anvil. Instead, we just pass normal gas values, and we handle the speed up/cancel, using purely the mining stop, and then mining a block after the tx is speed up/cancel

Reply:
Instead of `noMining` I wonder if the `--disable-block-gas-limit` setting is all you need. This would mean we could rely on auto mine and we'd be able to set gas to `0`.

Reply:
interesting suggestion! I've just tried and it seems that the transaction still fails. So maybe this flag enables to overpass the limit, but still fails with 0x0 values?

https://github.com/user-attachments/assets/6bcada58-829c-4219-a974-789ede173d35

Reply:
Ah! So, `disable-block-gas-limit` only disables the _maximum value_ check, not the lower check! There is another configuration option, `--block-base-fee-per-gas`, that when you set it to `0`, it secretly disables the lower limit check.

Normally a the "block base fee per gas" value varies with each new block; (for reasons), but setting it to `0` apparently turns off anvil's calculations as treats `0` as a constant, which is nice!

Reply:
ohh okay I didn't know about this one :raised_hands:  I've created an issue here and we can tackle it in a subsequent PR. 

I've also udpated the PR description with the 2 newly created tickets, so we can follow up the progress

https://github.com/MetaMask/MetaMask-planning/issues/4497